nixos/modules/system/boot/systemd.nix

```diff
@@ -520,7 +520,7 @@ in
       "systemd/system-shutdown" = { source = hooks "shutdown" cfg.shutdown; };
     });
 
-    services.dbus.enable = true;
+    services.dbus.enable = mkDefault true;
```

Comment:
I do not feel very good about `mkDefault` for a critical dependency like this. But I am not sure how else to implement this other than with `services.dbus.implementation = lib.mkOption { type = lib.mkEnum [ "dbus" "dbus-broker" ]; }`  which is not particularly nice either. So do not take this as NAK.

Reply:
The `mkDefault`s should be removed as well, then.

Reply:
And `mkIf cfg.enable` can now be moved to wrap the entire config:
```nix
  config = mkIf cfg.enable (mkMerge [
    {
      …
    }

    (mkIf (cfg.implementation == "dbus") {
      …
    })

    (mkIf (cfg.implementation == "dbus-broker") {
      …
    })
  ]);
```